presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -430,7 +408,8 @@ public QueryInfo createQuery(SessionContext sessionContext, String query)
             }
             if (statement instanceof Explain && ((Explain) statement).isAnalyze()) {
                 Statement innerStatement = ((Explain) statement).getStatement();
-                if (!(executionFactories.get(innerStatement.getClass()) instanceof SqlQueryExecutionFactory)) {
+                QueryType innerQueryType = STATEMENT_QUERY_TYPES.get(innerStatement.getClass());
+                if (innerQueryType == null || innerQueryType == QueryType.DATA_DEFINITION) {
```

Comment:
Maybe change the exception message to be

```
    throw new PrestoException(NOT_SUPPORTED, "EXPLAIN ANALYZE doesn't support statement type:  " + innerStatement.getClass().getSimpleName());
```

But I am not sure...

Reply:
I think the message can be updated per @wenleix 's suggestion.